### PR TITLE
Read zonemap functionality moved to common module

### PR DESCRIPTION
**Pull request by berland**

### CODE REVIEW

File: ecl2df/common.py [hunk 1]
```diff
@@ -621,3 +622,46 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
     del dframe["level_0"]
     dframe.index.name = ""
     return dframe
+
+
+def parse_zonemapfile(filename: str):
```
Comment (anders-kiaer) [inline ecl2df/common.py:627]:
```suggestion
def parse_zonemapfile(filename: str) -> Dict[int, str]:
```
...if this is what is returned?

File: ecl2df/common.py [hunk 2]
```diff
@@ -621,3 +622,46 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
...
+    """Return a dictionary from (int) K layers in the simgrid to strings
+
+    Typical usage is to map from grid layer to zone names.
+
+    The layer filename must currently follow format::
```
Comment (anders-kiaer) [inline ecl2df/common.py:632]:
```suggestion
    The layer filename must currently follow format:
```

File: ecl2df/common.py [hunk 3]
```diff
@@ -621,3 +622,46 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
...
+    zonelines = [line.split("--")[0] for line in zonelines]
+    zonelines = [line for line in zonelines if not line.startswith("#")]
+    zonelines = filter(len, zonelines)
+
+    zonemap = {}
+    for line in zonelines:
+        try:
+            linesplit = shlex.split(line)
+            map(str.strip, linesplit)
+            filter(len, linesplit)
```
Comment (anders-kiaer) [inline ecl2df/common.py:659]: Unused call?

File: ecl2df/common.py [hunk 4]
```diff
@@ -621,3 +622,46 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
...
+    zonelines = [line.strip() for line in zonelines]
+    zonelines = [line.split("--")[0] for line in zonelines]
+    zonelines = [line for line in zonelines if not line.startswith("#")]
+    zonelines = filter(len, zonelines)
+
+    zonemap = {}
+    for line in zonelines:
+        try:
+            linesplit = shlex.split(line)
+            map(str.strip, linesplit)
```
Comment (anders-kiaer) [inline ecl2df/common.py:658]: Unused call?

File: ecl2df/common.py [hunk 5]
```diff
@@ -621,3 +622,46 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
...
+            filter(len, linesplit)
+            (k_0, k_1) = "".join(linesplit[1:]).split("-")
+            for k_idx in range(int(k_0), int(k_1) + 1):
+                zonemap[k_idx] = linesplit[0]
```
Comment (anders-kiaer) [inline ecl2df/common.py:662]:
```suggestion
            zonemap.update(dict.fromkeys(range(int(k_0), int(k_1) + 1), linesplit[0]))
```
Maybe an alternative? Not tested.

File: ecl2df/common.py [hunk 6]
```diff
@@ -621,3 +622,46 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
...
+    Args:
+        filename (str): Absolute path to a zone map file (lyr format)
+
+    Returns:
+        dict, integer keys which are the K layers. Every layer mentioned
+            in the interval in the input file is present. Can be empty.
+    """
+    zonelines = Path(filename).read_text().splitlines()
+    zonelines = [line.strip() for line in zonelines]
+    zonelines = [line.split("--")[0] for line in zonelines]
```
Comment (anders-kiaer) [inline ecl2df/common.py:650]:
```suggestion
    zonelines = [line.strip().split("--")[0] for line in Path(filename).read_text().splitlines()]
```
Maybe more readable wrt. operations done?

File: ecl2df/common.py [hunk 7]
```diff
@@ -621,3 +622,46 @@ def stack_on_colnames(dframe, sep="@", stackcolname="DATE", inplace=True):
...
+    zonelines = [line for line in zonelines if not line.startswith("#")]
+    zonelines = filter(len, zonelines)
```
Comment (anders-kiaer) [inline ecl2df/common.py:652]:
```suggestion
    zonelines = [line for line in zonelines if line and not line.startswith("#")]
```

**Review by anders-kiaer — APPROVED** [review]
Some minor comments - otherwise LGTM.

File: ecl2df/common.py (hunk 2 above)
Comment (berland) [reply ecl2df/common.py:632]: The double colon is a formatting syntax, making the following paragraph formatted verbatim.

File: ecl2df/common.py (hunk 1 above)
Comment (berland) [reply ecl2df/common.py:627]: Will await this, until the whole of ecl2df is type hinted.

Comment (berland) [reply ecl2df/common.py:627]: https://github.com/equinor/ecl2df/issues/290

File: ecl2df/common.py (hunk 3 above)
Comment (berland) [reply ecl2df/common.py:659]: ah, yes it is unused! Good catch.

File: ecl2df/common.py (hunk 5 above)
Comment (berland) [reply ecl2df/common.py:662]: works perfect, thanks.

File: ecl2df/common.py (hunk 7 above)
Comment (berland) [reply ecl2df/common.py:652]: fixed

**Pull request merged**